test/Nerdbank.Streams.Tests/StreamPipeReaderTests.cs

```diff
@@ -0,0 +1,340 @@
+using System;
+using System.Buffers;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Nerdbank.Streams;
+using NSubstitute;
+using Xunit;
+
+namespace Nerdbank.Streams.UnitTests
+{
+    /// <summary>
+    /// Unit tests for the <see cref="StreamPipeReader"/> class.
+    /// </summary>
+    public class StreamPipeReaderTests
+    {
+        private readonly byte[] testData = new byte[] { 1, 2, 3, 4, 5 };
+
+        /// <summary>
+        /// A custom stream that is not readable to test constructor validation.
+        /// </summary>
+        private class NonReadableStream : MemoryStream
+        {
+            public override bool CanRead => false;
+        }
+
+        /// <summary>
+        /// A custom stream that introduces a delay in Read and ReadAsync calls to simulate blocking behavior.
+        /// </summary>
+        private class DelayingStream : MemoryStream
+        {
+            private readonly int delayMilliseconds;
+
+            public DelayingStream(byte[] buffer, int delayMilliseconds) : base(buffer)
+            {
+                this.delayMilliseconds = delayMilliseconds;
+            }
+
+            //             public override async Task<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default) [Error] (40-45)CS0508 'StreamPipeReaderTests.DelayingStream.ReadAsync(Memory<byte>, CancellationToken)': return type must be 'ValueTask<int>' to match overridden member 'MemoryStream.ReadAsync(Memory<byte>, CancellationToken)'
+            //             {
+            //                 await Task.Delay(delayMilliseconds, cancellationToken);
+            //                 return await base.ReadAsync(buffer, cancellationToken);
+            //             }
+
+
+            public override int Read(Span<byte> buffer)
+            {
+                Thread.Sleep(delayMilliseconds);
```

Comment:
This is peculiar. I don't usually care for Thread.Sleep in tests because it makes tests run slowly. 
In this case, this private class isn't even referenced by anything. So why was it generated?